app/src/androidTest/java/com/github/epfl/meili/MainActivityTest.kt

```diff
@@ -1,42 +1,73 @@
+import androidx.test.espresso.Espresso.pressBack
 import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.action.ViewActions.click
 import androidx.test.espresso.assertion.ViewAssertions
+import androidx.test.espresso.assertion.ViewAssertions.matches
 import androidx.test.espresso.intent.Intents
 import androidx.test.espresso.intent.matcher.IntentMatchers.toPackage
 import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.matcher.ViewMatchers.*
 import androidx.test.ext.junit.rules.ActivityScenarioRule
 import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.schibsted.spain.barista.interaction.PermissionGranter
 import org.junit.Rule
 import org.junit.Test
+import org.junit.Before
+import org.junit.After
 import org.junit.runner.RunWith
 
 @RunWith(AndroidJUnit4::class)
 class MainActivityTest {
     @get:Rule
-    var testRule: ActivityScenarioRule<MainActivity?>? = ActivityScenarioRule(
-            MainActivity::class.java
-    )
+    var testRule: ActivityScenarioRule<MainActivity> = ActivityScenarioRule(MainActivity::class.java)
+
+    @Before
+    fun setup() {
+        Intents.init()
+    }
+
+    @After
+    fun cleanup() {
+        Intents.release()
+    }
 
     @Test
     fun clickingOnSignInViewButtonShouldLaunchIntent() {
-        Intents.init()
-        Espresso.onView(ViewMatchers.withId(R.id.launchSignInView))
-                .check(ViewAssertions.matches(ViewMatchers.isClickable())).perform(ViewActions.click())
+        onView(withId(R.id.launchSignInView))
+            .check(matches(isClickable())).perform(click())
 
         Intents.intended(toPackage("com.github.epfl.meili"))
-        Intents.release()
     }
 
     @Test
     fun clickingOnChatViewButtonShouldLaunchIntent() {
-        Intents.init()
-        Espresso.onView(ViewMatchers.withId(R.id.launchChatView))
-            .check(ViewAssertions.matches(ViewMatchers.isClickable())).perform(ViewActions.click())
+        onView(withId(R.id.launchChatView))
+            .check(matches(isClickable())).perform(click())
+
+        Intents.intended(toPackage("com.github.epfl.meili"))
+    }
+
+    @Test
+    fun clickingOnMapViewButtonShouldLaunchIntent() {
+        onView(withId(R.id.launchMapView))
+            .check(matches(isClickable())).perform(click())
 
         Intents.intended(toPackage("com.github.epfl.meili"))
-        Intents.release()
     }
 
+//    @Test
+//    fun testNavigation() {
+//        PermissionGranter.allowPermissionsIfNeeded("android.permissions.ACCESS_FINE_LOCATION")
+//        onView(withId(R.id.launchSignInView)).perform(click())
+//        onView(withId(R.id.signInButton)).check(matches(isDisplayed()))
+//        pressBack()
+//        onView(withId(R.id.launchMap)).perform(click())
+//        onView(withId(R.id.map)).check(matches(isDisplayed()))
+//        pressBack()
+//        onView(withId(R.id.launchSignInView)).check(matches(isClickable())).perform(click())
```

Comment:
It is commented on purpose. This is because the test works on a device (where no permissions are given by default) but doesn't work with gradlew as the permission dialog box doesn't appear. I suppose this is because the permission is given  by default and I haven't found a way to revoke permissions at the start of each test.

Reply:
I've added this comment to the code too.

Reply:
Okay Thanks!